test/standalone/Vegetation/plant_hydraulics_test.jl

```diff
@@ -283,70 +281,110 @@ for FT in (Float32, Float64)
                 atmos = atmos,
                 radiation = radiation,
             )
-            # Set system to hydrostatic equilibrium state by setting fluxes to zero, and setting LHS of both ODEs to 0
-            function initial_compute_exp_tendency!(F, Y)
-                AI = (; leaf = LAI(1.0), root = RAI, stem = SAI)
-                T0A = FT(1e-8) * AI[:leaf]
-                for i in 1:(n_leaf + n_stem)
-                    if i == 1
-                        fa =
-                            sum(
-                                water_flux.(
-                                    root_depths,
+            # Create a system of equations for a given rooting_depth for each cell
+            function create_exp_tendency_func(rooting_depth)
+                # Solve for hydrostatic equilibrium
+                return function initial_compute_exp_tendency!(F, Y)
+                    AI = (; leaf = LAI(1.0), root = RAI, stem = SAI)
+                    T0A = FT(1e-8) * AI[:leaf]
```

Comment:
`FT` here should probably not be grabbed from the outer-scope, as that will result in a captured variable, and that can lead to type instabilities (which can hurt performance).

Reply:
I'm not sure how this test could be rewritten to not use variables from the outer scope. Would passing in the outer scope variables as arguments fix the issue?

`function create_exp_tendency_func(rooting_depth, LAI, FT)
                # Solve for hydrostatic equilibrium
                    function initial_compute_exp_tendency!(F, Y)
                        AI = (; leaf = LAI(1.0), root = RAI, stem = SAI)
                        T0A = FT(1e-8) * AI[:leaf]`

When I check using @code_warntype, both create_exp_tendency_func and the function it returns are type stable.  Doesn't the inner function still capture the variables FT and LAI though?

Reply:
I think in this case it is fine to use variables from the outer scope, as this function is run 1x, and in a test where performance is not important.